lib/broccoli/ember-app.js

```diff
@@ -669,54 +693,89 @@ EmberApp.prototype._filterAppTree = function() {
     return this._cachedFilterAppTree;
   }
 
-  var podPatterns = this._podTemplatePatterns();
-  var excludePatterns = podPatterns.concat([
-    // note: do not use path.sep here Funnel uses
-    // walk-sync which always joins with `/` (not path.sep)
+  var appExcludePatterns = [
     'styles/**/*',
     'templates/**/*',
-  ]);
-
-  this._cachedFilterAppTree = new Funnel(this.trees.app, {
-    exclude: excludePatterns,
-    annotation: 'Funnel: Filtered App'
-  });
+  ];
+  this._cachedFilterAppTree = this._filterAnyAppTree(this.trees.app, appExcludePatterns, 'App');
 
   return this._cachedFilterAppTree;
 };
 
-EmberApp.prototype.podTemplates = function() {
-  return new Funnel(this.trees.app, {
+/**
+  Filters templates from the `fastboot/app` tree.
+
+  @private
+  @method _filterFastbootAppTree
+  @return {Tree}
+*/
+EmberApp.prototype._filterFastbootAppTree = function() {
+  if (this._cachedFilterFastbootAppTree) {
+    return this._cachedFilterFastbootAppTree;
+  }
+
+  if (this.trees.fastbootApp) {
+    var appExcludePatterns = [
```

Comment:
Pretty sure it's probably best to still include `styles` here. Possible that a JS file sneaks in that folder and we have a bad day.